src/NovelRT/DotNet/RuntimeService.cpp

```diff
@@ -85,7 +85,7 @@ namespace NovelRT::DotNet {
       if (result != static_cast<int>(HostApiBufferTooSmall))
       {
         _logger.logError("Failed to locate hostfxr: {}", result);
-        throw std::runtime_error("Failed to locate hostfxr");
+        throw Exceptions::OutOfMemoryException("Failed to locate hostfxr");
```

Comment:
I'm wondering if this is really the correct message here. I imagine that a standard out of memory exception message would be enough here. We could consider acceping an "inner" reason, in this case the host api buffer being too small to get the location.